hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -222,7 +223,16 @@ public long gasRequirement(final Bytes bytes) {
     @Override
     public PrecompileContractResult computePrecompile(final Bytes input, @NonNull final MessageFrame frame) {
         prepareFields(frame);
-        prepareComputation(input, updater::unaliased);
+        try {
+            prepareComputation(input, updater::unaliased);
+        } catch (InvalidTransactionException e) {
+            if (NOT_SUPPORTED.equals(e.getResponseCode())) {
```

Comment:
possibly: compute the return code in a separate statement to be used in the next statement to create the result

```
final var haltReason = NOT_SUPPORTED.equals(e.getResponseCode()) ? HederaExceptionalHaltReason.NOT_SUPPORTED : HederaExceptionalHaltReason.ERROR_DECODING_PRECOMPILE_INPUT;
return PrecompileContractResult.halt(null, Optional.of(haltReason);
```

Reply:
yeah fixed.